### PR TITLE
Updated slider preview to work with responsive sizing

### PR DESCRIPTION
**Pull request by jessipedia**
Will fix issue 102 on the `rt-dash`

This:
- changes the D3 joins so that styling isn't overwritten on updates
- checks to see if `onUpdateCallback` is present on the slider and adds it if missing (perviously it was lost when the chart re-rendered)

### CODE REVIEW

File: rickshaw.js [hunk 1]
```diff
@@ -3215,6 +3222,18 @@ Rickshaw.Graph.RangeSlider.Preview = Rickshaw.Class.create({
     // Use the first graph as the "master" for the frame state
     var masterGraph = this.graphs[0]
 
+    var funcArray = masterGraph.updateCallbacks.map(function(func) {
+      return func.name === 'onUpdateCallback'
```
Comment (jessipedia) [inline rickshaw.js]: This does not seem like an ideal way to check if the function is present. Is there a better way? Moving `onUpdateCallback` into `render`'s scope and checking `includes()` with the function directly produced a lot of false negatives.

File: rickshaw.js [hunk 2]
```diff
@@ -3172,8 +3182,7 @@ Rickshaw.Graph.RangeSlider.Preview = Rickshaw.Class.create({
       var graph = new Rickshaw.Graph(graphArgs)
       self.previews.push(graph)
 
-      parent.onUpdate(function() {
-        graph.render()
+      parent.onUpdate(function(){
```
Comment (evanemolo) [inline rickshaw.js]: `parent.onUpdate(self.render)`

File: src/js/Rickshaw.Graph.RangeSlider.Preview.js [hunk 3]
```diff
@@ -137,10 +148,7 @@ Rickshaw.Graph.RangeSlider.Preview = Rickshaw.Class.create({
       var graph = new Rickshaw.Graph(graphArgs)
       self.previews.push(graph)
 
-      parent.onUpdate(function() {
-        graph.render()
-        self.render()
-      })
+      parent.onUpdate(self.render())
```
Comment (evanemolo) [inline src/js/Rickshaw.Graph.RangeSlider.Preview.js]: No bueno - this is going to immediately invoke `render`, as you're now passing the *invoked* render function to `onUpdate` - this should be: `parent.onUpdate(self.render)`

Comment (jessipedia) [reply src/js/Rickshaw.Graph.RangeSlider.Preview.js]: Fixed for real now.

**Pull request merged**